doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,613 @@
+
+***Hot update : The parameter can be updated on the fly. This feature should be patched on wpa_supplicant***
+
+##### 3.4.1.3 Enable MACsec
+
+``` bash
+# Enable SONiC MACsec driver at the port. It is the physical port in the interface stack
+wpa_cli -g{{DOMAIN_SOCK}} interface_add {{PORT}} {{CONFNAME}} macsec_sonic
+
+# Add a network at the port and get the Network Id according from this instruction
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} add_network
+
+# Set Key management. Use preshared key
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{NETWORK_ID}} key_mgmt NONE
+
+# Set IEEE 802.1X/EAPOL options. When using wired authentication (including MACsec drivers), eapol_flags must be set to 0.
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{NETWORK_ID}} eapol_flags 0
+
+# Set the macsec policy. The value 1 means Should secure, accept key server's advice to determine whether to use a secure session or not.
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{NETWORK_ID}} macsec_policy 1
+
+# Set the transmit mode
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{NETWORK_ID}} macsec_integ_only {{ENABLE_ENCRYPT}}
+
+# Set the Secure Connectivity Association Key
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{network_id}} mka_cak {{CAK}}
+
+# Set the CAK Name
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{network_id}} mka_ckn {{CKN}}
+
+# Set the priority of MKA actor
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{network_id}} mka_priority {{PRIORITY}}
+
+# Set replay protect flag
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{network_id}} macsec_replay_protect {{ENABLE_REPLAY_PROTECT}}
+
+# Set replay protect window size
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} set_network {{network_id}} macsec_replay_window {{REPLAY_WINDOW}}
+
+# Enable the network
+wpa_cli -g{{DOMAIN_SOCK}} IFNAME={{PORT}} enable_network {{network_id}}
+```
+
+##### 3.4.1.4 Disable MACsec
+
+``` bash
+# Disable MACsec driver at the port
+wpa_cli -g{{DOMAIN_SOCK}} interface_remove {{PORT}}
+```
+
+#### 3.4.2 WPA Supplicant
+
+Wpa_supplicant(<https://salsa.debian.org/debian/wpa>) is Wi-Fi Protected Access client and IEEE 802.1X(MACsec) supplicant. A single instance of wpa_supplicant is able to manage multiple MACsec channels. Wpa_supplicant will be managed by supervisor in the macsec container as a daemon. Start command is `wpa_supplicant -Dmacsec_sonic -g{{DOMAIN_SOCK}}`.
+The first parameter, macsec_sonic, indicates that the SONiC MACsec plugin is used.This parameter should be declared at SONiC MACsec Plugin.
+The second parameter, DOMAIN_SOCK, indicates the position of domain sock, that will be used to interact with wpa_cli.
+The details of SONiC MACsec plugin will be mentioned at next chapter.
```

Comment:
what is referring to next chapter?, do you mean, next phase?

Reply:
Done, please review it.